modules/native.nix

```diff
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.typelevelShell.native;
+in
+{
+  options.typelevelShell.native = {
+    enable = mkEnableOption "Provide scala-native environment";
+
+    libraries = mkOption {
+      type = types.listOf types.package;
+      default = [ pkgs.zlib ];
+      description = "A list of split-output libraries to include (dev output) and link (lib output)";
```

Comment:
I don't know whether this is a good assumption.  It's probably almost always right, and prevents Nix+C rookies from learning about `getDev` and `getLib`.

I don't know anything about Scala Native.  How often might we want to link against a library without headers, or include a header-only library?  Those would _not_ go in this key, and if it's common, this key shouldn't exist.

Reply:
@armanbilge Do you have a sense of when we'd want includes xor libraries, not includes and libraries?

Reply:
> How often might we want to link against a library without headers

This would be quite strange. Usually headers define what's the public API (and what's not). Although Scala Native is capable of consuming libraries without the headers.

> or include a header-only library

Header-only libraries are not uncommon, but they are difficult to work with from Scala Native since it can't actually consume headers. You end up having to write "glue code" in C to expose them.

Reply:
These cases can be covered by `(getLib pkg.blah)` or `(getDev pkg.blah)` in `packages`.  The only reason for this setting is a convenience, because usually we'll want both per blah.